### PR TITLE
MB-13461 submit report violations

### PR DESCRIPTION
**Pull request by leia-sefkin**
## Summary

**Acceptance Criteria **

- Given a QAE that is filling out an evaluation report with violations using MilMove 
- When they are ready to submit the report 
- Then the MilMove system is able to process their submission if the violations section of the report is in a valid state

This story implements report submission when violations are in a valid state. Submission has already been implemented from the standard report preview page, so much of the code is copied over. I am considering how to refactor, input and ideas are more than welcome. 

## Setup to Run Your Code

- make db_dev_fresh (new migrations)
- make swagger_generate (latest swagger)
- make server_run (backend)
- make office_client_run (office client)

## Verification Steps for Reviewers

- login or create new qae csr user
- go to the evaluation reports page for a move (or the test data gen'd EVLRPT)
- edit or create a new report and select yes for "violations observed" on the evaluation counseling form 
- click the next button to select violations, set serious incident to no [not sure if this is in scope for this story] and click review and submit
- submit from the report preview page

- [ ] the review and submit buttons are not enabled until the form is in a valid state
- [ ] submission sets the form to submitted/view only
- [ ] latest violation data is saved to the report ahead of submission

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.jsx [hunk 1]
```diff
@@ -116,195 +173,230 @@ const EvaluationViolationsForm = ({ violations, evaluationReport, reportViolatio
...
+  const handlePreviewReport = async (values) => {
+    // save updates
+    await saveDraft(values);
+
+    // open the modal to submit
+    setIsSubmitModalOpen(!isSubmitModalOpen);
+  };
+
   return (
-    <Formik
```
Comment (leia-sefkin) [inline src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.jsx:120]: these diffs make it look like more was changed than it was - things just got shifted around but this is all still there

File: src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.jsx [hunk 2]
```diff
@@ -1,24 +1,26 @@
-import React from 'react';
```
Comment (leia-sefkin) [inline src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.jsx:1]: diffs on this file make it look like a lot was changed that wasn't - recommend viewing with the "hide whitespace changes" mode on

File: src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.test.jsx [hunk 3]
```diff
@@ -60,23 +154,25 @@ jest.mock('react-router', () => ({
...
     evaluationReport: mockEvaluationReport,
     reportViolations: [mockReportViolation],
     violations: [mockViolation],
+    customerInfo,
+    mtoShipments: mockShipmentData,
   };
 
   return render(
-    <MockProviders initialEntries={[`/moves/${mockMoveCode}/evaluation-reports/${mockReportId}`]}>
+    <MockProviders initialEntries={[`/moves/A12345/evaluation-reports/db30c135-1d6d-4a0d-a6d5-f408474f6ee2`]}>
```
Comment (leia-sefkin) [inline src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.test.jsx:175]: jest was complaining about the initialization of these values so i had to hard code them

File: src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.test.jsx [hunk 4]
```diff
@@ -185,11 +272,13 @@ describe('EvaluationViolationsForm', () => {
       expect(screen.getByText('Observed pickup spread end date')).toBeInTheDocument();
     });
   });
+});
 
+describe('EvaluationViolationsForm Buttons', () => {
   it('re-routes back to the eval report', async () => {
     renderForm();
     // Click back button
-    await userEvent.click(await screen.findByRole('button', { name: '< Back to Evaluation form' }));
```
Comment (leia-sefkin) [inline src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.test.jsx:192]: removed extra awaits

File: src/pages/Office/EvaluationViolations/EvaluationViolations.jsx [hunk 5]
```diff
@@ -8,14 +8,14 @@ import styles from '../TXOMoveInfo/TXOTab.module.scss';
 
 import evaluationViolationsStyles from './EvaluationViolations.module.scss';
 
-import { useEvaluationReportQueries, usePWSViolationsQueries } from 'hooks/queries';
+import { useEvaluationReportShipmentListQueries, usePWSViolationsQueries } from 'hooks/queries';
```
Comment (leia-sefkin) [inline src/pages/Office/EvaluationViolations/EvaluationViolations.jsx:11]: updated the query so we can send shipment info to the preview modal

**Comment by Ryan-Koch** [comment]
This is looking good! Only noticed two things:
 1. It looks like we're losing the customer info when I play with it locally and go from the violations page to the preview modal. 
 2. It looks like we're not passing the description text along to save when that action occurs before the modal opens (I suspect we missed this when reviewing the save draft stuff for this one). 
 
I suspect we'd be best served by breaking that second issue off into its own thing. I'd also be open to doing the same with the first if it doesn't seem like something that's easy to address.

File: src/components/Office/CustomerSupportRemarkText/CustomerSupportRemarkText.module.scss [hunk 6]
```diff
@@ -13,6 +13,7 @@
 .customerSupportRemarkText {
   @include u-margin-bottom(0.5);
   @include u-margin-top(0);
+  overflow-wrap: anywhere;
```
Comment (leia-sefkin) [inline src/components/Office/CustomerSupportRemarkText/CustomerSupportRemarkText.module.scss:16]: small bug fix for CSR remarks which need to wrap (e.g. a long url link)

File: src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.jsx [hunk 7]
```diff
@@ -107,7 +107,42 @@ const EvaluationViolationsForm = ({ violations, evaluationReport, reportViolatio
...
+    }),
+    observedPickupSpreadStartDate: Yup.date().when('kpiViolations', {
+      is: (kpiViolations) => kpiViolations.includes('observedPickupSpreadDates'),
+      then: Yup.date().required(),
+    }),
+    observedPickupSpreadEndDate: Yup.date().when('kpiViolations', {
+      is: (kpiViolations) => kpiViolations.includes('observedPickupSpreadDates'),
+      then: Yup.date().required(),
+    }),
+  });
```
Comment (PDickeyTruss) [inline src/components/Office/EvaluationViolationsForm/EvaluationViolationsForm.jsx:145]: This validation looks correct to me and works as expected in the app 🎉

File: src/components/Office/CustomerSupportRemarkText/CustomerSupportRemarkText.module.scss (hunk 6 above)
Comment (PDickeyTruss) [reply src/components/Office/CustomerSupportRemarkText/CustomerSupportRemarkText.module.scss:16]: Fix worked as expected, long word now wraps ✅

**Review by PDickeyTruss — APPROVED** [review]
Looks great to me code-wise and works as expected. Also includes some nice cleanup witch all looks good as well. Approved! 🚢

**Comment by PDickeyTruss** [comment]
Looking at Ryan's comment: 

> It looks like we're not passing the description text along to save when that action occurs before the modal opens (I suspect we missed this when reviewing the save draft stuff for this one).

Looks like this is a bug intorduced in this PR. The issue is line 390 of EvaluationViolatinosForm.jsx renames the text area to `yesSeriousIncident` but it is expecting it to be named `seriousIncidentDesc` when pulling out the values (line 171). Will probably want to fix this before merging in, should be easy to clean up 👍

**Review by cosburn — APPROVED** [review]
Looks good to me!

**Pull request merged**